metricbeat/module/kafka/partition/partition.go

```diff
@@ -163,16 +164,20 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) {
 				}
 
 				// TODO (deprecation): Remove fields from MetricSetFields moved to ModuleFields
-				r.Event(mb.Event{
+				sent := r.Event(mb.Event{
 					ModuleFields: common.MapStr{
 						"broker": evtBroker,
 						"topic":  evtTopic,
 					},
 					MetricSetFields: event,
 				})
+				if !sent {
+					return errors.New("metricset is closed")
```

Comment:
Should we report an error here? I understood this is `false` when the metricset is stopped, which is not an error but a known condition

Reply:
@exekias I think you are correct: https://github.com/elastic/beats/issues/11666

Reply:
Yah. In theory you _don't_ need to return an error, but for whatever reason `return nil` just seemed...weird here.

Reply:
We should `return nil` here, yes. If you want you can log something like "Fetch interrupted, metricset stopping" at the debug level.